pkg/termination/termination.go

```
@@ -58,6 +65,47 @@ func NewHandler(logger logr.Logger, cfg *rest.Config, pollInterval time.Duration
+	if err != nil {
+		return nil, fmt.Errorf("error parsing poll URL: %v", err)
+	}
+
+	logger.V(1).Info("Using API version 2020-07-01, SpotRebalanceRecommendation is not available")
+	return pollURL, nil
+}
+
+// endpointAvailable checks if the endpoint is available by making a GET request to the endpoint.
+// When an API version is not available, the endpoint will return a 404.
+func endpointAvailable(pollURL *url.URL) bool {
+	req, err := http.NewRequest("GET", pollURL.String(), nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)